File: apps/spring-boot-api/src/main/java/org/bootstrapbugz/api/admin/service/AdminService.java
package org.bootstrapbugz.api.admin.service;

import java.util.List;
import org.bootstrapbugz.api.admin.payload.request.AdminRequest;
import org.bootstrapbugz.api.admin.payload.request.ChangeRoleRequest;
import org.bootstrapbugz.api.user.payload.response.UserResponse;

public interface AdminService {
  List<UserResponse> findAllUsers();

  void changeRole(ChangeRoleRequest changeRoleRequest);

  void lock(AdminRequest adminRequest);

  void unlock(AdminRequest adminRequest);

  void activate(AdminRequest adminRequest);

  void deactivate(AdminRequest adminRequest);

  void delete(AdminRequest adminRequest);
}

File: libs/angular-ui/admin/src/lib/admin.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AdminRequest, ChangeRoleRequest, UserResponse } from '@bootstrapbugz/shared';

@Injectable({
  providedIn: 'root',
})
export class AdminService {
  API_URL = 'http://localhost:8181/v1.0/admin';

  constructor(private http: HttpClient) {}

  findAllUsers() {
    return this.http.get<UserResponse>(`${this.API_URL}/users`);
  }

  activate(adminRequest: AdminRequest) {
    return this.http.put<void>(`${this.API_URL}/users/activate`, adminRequest);
  }

  deactivate(adminRequest: AdminRequest) {
    return this.http.put<void>(`${this.API_URL}/users/deactivate`, adminRequest);
  }

  lock(adminRequest: AdminRequest) {
    return this.http.put<void>(`${this.API_URL}/users/lock`, adminRequest);
  }

  unlock(adminRequest: AdminRequest) {
    return this.http.put<void>(`${this.API_URL}/users/unlock`, adminRequest);
  }

  delete(adminRequest: AdminRequest) {
    return this.http.request<void>('delete', `${this.API_URL}/users/delete`, {
      body: adminRequest,
    });
  }

  changeRole(changeRoleRequest: ChangeRoleRequest) {
    return this.http.post<void>(`${this.API_URL}/users/role`, changeRoleRequest);
  }
}
